cmake/findDependencies.cmake

```diff
@@ -7,12 +7,19 @@ if (BUILD_GUI)
 endif()
 
 if (HAVE_RULES)
-    find_library(PCRE pcre)
-    if (NOT PCRE)
+    find_library(PCRE_LIBRARY pcre)
+    if (NOT PCRE_LIBRARY)
         message(FATAL_ERROR "pcre dependency for RULES has not been found")
     endif()
 endif()
 
+if (USE_Z3)
+    find_library(Z3_LIBRARY z3)
+    if (NOT Z3_LIBRARY)
+        message(FATAL_ERROR "z3 dependency has not been found")
+    endif()
+endif()
```

Comment:
On the Fedora Linux system here, I need the following in addition, so that `z3++.h` is found:
```diff
diff --git a/cmake/findDependencies.cmake b/cmake/findDependencies.cmake
index 70cfdd8c5..ac88ed354 100644
--- a/cmake/findDependencies.cmake
+++ b/cmake/findDependencies.cmake
@@ -18,6 +18,11 @@ if (USE_Z3)
     if (NOT Z3_LIBRARY)
         message(FATAL_ERROR "z3 dependency has not been found")
     endif()
+    find_path(Z3_INCLUDE z3++.h PATH_SUFFIXES "z3")
+    if (NOT Z3_INCLUDE)
+        message(FATAL_ERROR "z3++.h has not been found")
+    endif()
+    include_directories(${Z3_INCLUDE})
 endif()
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
```

Error otherwise:
```console
cppcheck/lib/exprengine.cpp:144:10: fatal error: z3++.h: No such file or directory
  144 | #include <z3++.h>
      |          ^~~~~~~~
```

OS: Fedora 31, x86_64
z3: version 4.8.7

Reply:
This should not be necessary if you use `find_package(Z3)` and the `z3::libz3` target.

Reply:
`find_package(Z3)` cannot be used yet under Fedora, because the rpm spec file does not install the required z3 .cmake file so far. This will change - sooner or later ...

Reply:
Thanks, I will re-add the `find_include()` for both the external libraries.